File: 0290-word-pattern/0290-word-pattern.cpp
class Solution {
public:
    bool wordPattern(string pattern, string s) {
        
                unordered_map<char,string>gzip;
                unordered_map<string,char>gzip2;
                stringstream ss(s);
                int i=0;
                string word;
                int cnt=0;
                while(ss>>word){
                    
                if((gzip.find(pattern[i])!=gzip.end() and gzip[pattern[i]]!=word) or (gzip2.find(word)!=gzip2.end() and  gzip2[word]!=pattern[i])){
                                    return false;
                    }
                    
                    gzip[pattern[i]] = word;
                    gzip2[word]=pattern[i];
                    i++;  
                    cnt++;
                }
            
                if(cnt!=pattern.size())
                        return false;
                
                return true;
        
    }

    
};

File: 0290-word-pattern/0290-word-pattern.py
class Solution:
    
    
    def wordPattern(self, pattern: str, s: str) -> bool:
    
                
            anim_list=s.split(' ')
            dict1={}
            dict2={}
            
            if len(anim_list)!=len(pattern):
                    return False
            
            for item1,item2 in zip(pattern,anim_list):
            
                    if item1 not in dict1:
                            dict1[item1]=item2
                    else:
                            if dict1[item1]!=item2:
                                    return False
                                
                    if item2 not in dict2:
                            dict2[item2]=item1
                    else:
                            if dict2[item2]!=item1:
                                    return False
                                
                            
            return True